CMakeLists.txt

```diff
@@ -435,114 +480,84 @@ function(is_sdk_requested name result_var_name)
   endif()
 endfunction()
 
-# FIXME: separate the notions of SDKs used for compiler tools and target
-# binaries.
-if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
-  set(CMAKE_EXECUTABLE_FORMAT "ELF")
+# Required from this point:
+# - SWIFT_HOST_VARIANT_SDK  - OS of system which will host the compiler
+# - SWIFT_HOST_VARIANT_ARCH - arch of system which will host the compiler
+# 
+# FIXME: cross-compile standard libarary targets are specified by "SDK",
+# which locks each one to the hardcoded architectures defined here (e.g. armv7 for ANDROID)
```

Comment:
I mean that the parameters we specify in `SWIFT_SDKS` are lacking architecture specifics, so we need to hard-code it. For example, the SDK for Android is just `ANDROID`, which we assume below to be armv7. The iOS SDKs all have their architectures hardcoded because they are just specified by name (e.g. `IOS` or `WATCHOS`). We can't cross-compile the standard library for another linux architecture, because the SDK list would just be `LINUX` and we couldn't disambiguate it from the host.

To fix it, we would need to append the architecture to the SDK, for example: `OSX-x86_64;IOS-armv7`. We could easily do that - we have all of that information in `build-script-impl`.